src/mesh/msh_gmshreader.cc

```diff
@@ -199,86 +203,87 @@ void GmshMeshReader::read_elements(Tokenizer &tok, Mesh * mesh) {
 
 
 
-void GmshMeshReader::read_physical_names(Tokenizer &tok, Mesh * mesh) {
+void GmshMeshReader::read_physical_names(Mesh * mesh) {
+	ASSERT( mesh , "Argument mesh is NULL.\n");
     using namespace boost;
 
-    if (! tok.skip_to("$PhysicalNames", "$Nodes") ) return;
+    if (! tok_.skip_to("$PhysicalNames", "$Nodes") ) return;
     try {
-        tok.next_line(false);
-        unsigned int n_physicals = lexical_cast<unsigned int> (*tok);
-        ++tok; // end of line
+    	tok_.next_line(false);
+        unsigned int n_physicals = lexical_cast<unsigned int> (*tok_);
+        ++tok_; // end of line
 
         for (unsigned int i = 0; i < n_physicals; ++i) {
-            tok.next_line();
+        	tok_.next_line();
             // format of one line:
             // dim    physical-id    physical-name
 
-            unsigned int dim = lexical_cast<unsigned int>(*tok); ++tok;
-            unsigned int id = lexical_cast<unsigned int>(*tok); ++tok;
-            string name = *tok; ++tok;
+            unsigned int dim = lexical_cast<unsigned int>(*tok_); ++tok_;
+            unsigned int id = lexical_cast<unsigned int>(*tok_); ++tok_;
+            string name = *tok_; ++tok_;
 
-            mesh->region_db_.add_region(id, name, dim);
+            mesh->region_db_.add_region(id, name, dim, "defined in '$PhysicalNames' section");
```

Comment:
Store just the input address without wrapping text. in this case form a special address:
(mesh: <file>, $PhysicalNames)